src/hdmf/common/resources.py

```diff
@@ -165,6 +166,59 @@ def __init__(self, **kwargs):
         self.object_keys = kwargs['object_keys'] or ObjectKeyTable()
         self.type_map = kwargs['type_map'] or get_type_map()
 
+    @staticmethod
+    def assert_external_resources_equal(left, right, check_dtype=True):
+        """
+        Compare that the keys, resources, entities, objects, and object_keys tables match
+
+        :param left: ExternalResources object to compare with right
+        :param right: ExternalResources object to compare with left
+        :param check_dtype: Enforce strict checking of dtypes. Dtypes may be different
+            for example for ids, where depending on how the data was saved
+            ids may change from int64 to int32. (Default: True)
+        :returns: The function returns True if all values match. If mismatches are found,
+            AssertionError will be raised.
+        :raises AssertionError: Raised if any differences are found. The function collects
+            all differences into a single error so that the assertion will indicate
+            all found differences.
+        """
+        errors = []
+        try:
+            pd.testing.assert_frame_equal(left.keys.to_dataframe(),
+                                          right.keys.to_dataframe(),
+                                          check_dtype=check_dtype)
+        except AssertionError as e:
+            errors.append(e)
+        try:
+            pd.testing.assert_frame_equal(left.objects.to_dataframe(),
+                                          right.objects.to_dataframe(),
+                                          check_dtype=check_dtype)
+        except AssertionError as e:
+            errors.append(e)
+        try:
+            pd.testing.assert_frame_equal(left.resources.to_dataframe(),
+                                          right.resources.to_dataframe(),
+                                          check_dtype=check_dtype)
+        except AssertionError as e:
+            errors.append(e)
+        try:
+            pd.testing.assert_frame_equal(left.entities.to_dataframe(),
+                                          right.entities.to_dataframe(),
+                                          check_dtype=check_dtype)
+        except AssertionError as e:
+            errors.append(e)
+        try:
+            pd.testing.assert_frame_equal(left.object_keys.to_dataframe(),
+                                          right.object_keys.to_dataframe(),
+                                          check_dtype=check_dtype)
+        except AssertionError as e:
+            errors.append(e)
```

Comment:
It looks like this assertion isn't being triggered. Could you check the test for this? @mavaylon1

Reply:
The tests pass for each case. I'm not sure what you mean.

Reply:
The tests are passing but that line is not being run by the tests. You can check the code coverage here: https://app.codecov.io/gh/hdmf-dev/hdmf/pull/799 or by running:
```
pytest --cov
coverage html && open ./tests/coverage/htmlcov/index.html
```

Reply:
the corresponding test for this should be ``test_invalid_object_keys_assert_external_resources_equal``. Probably a bug in how the test is setting up the data.